EIPS/eip-4524.md

```diff
@@ -0,0 +1,82 @@
+
+## Specification
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+In order to be compliant with this EIP, and ERC20-compliant contract MUST also implement the following functions:
+```solidity
+pragma solidity 0.8.10;
+
+import './IERC20.sol';
+import './IERC165.sol';
+
+// the EIP-165 interfaceId for this interface is 0x534f5876
+
+interface SaferERC20 is IERC20, IERC165 {
+  function safeTransfer(address to, uint256 amount) external returns(bool);
+  function safeTransfer(address to, uint256 amount, bytes memory data) external returns(bool);
+  function safeTransferFrom(address from, address to, uint256 amount) external returns(bool);
+  function safeTransferFrom(address from, address to, uint256 amount, bytes memory data) external returns(bool);
+}
+```
+`safeTransfer` and `safeTransferFrom` MUST transfer as expected to EOA addresses, and to contracts implementing `ERC20Receiver` and returning the function selector (`0x4fc35859`) when called, and MUST revert when transferring to a contract which either does not have `ERC20Receiver` implemented, or does not return the function selector when called.
+
+In addition, a contract accepting safe transfers MUST implement the following if it wishes to accept safe transfers, and MUST return the function selector (`0x4fc35859`):
+```solidity
+pragma solidity 0.8.10;
+
+import './IERC165.sol';
+
+interface ERC20Receiver is IERC165 {
+  function onERC20Received(
+    address _operator,
+    address _from,
+    uint256 _amount,
+    bytes _data
+  ) external returns(bytes4);
+}
+```
+
+## Rationale
+
+This EIP is meant to be minimal and straightforward. Adding EIP-165 to ERC20 is useful for a number of applications, and outside of a minimal amount of code increasing contract size, carries no downside. The `safeTransfer` and `safeTransferFrom` functions are well recognized from ERC721 and ERC1155, and therefore keeping identical naming conventions is reasonable, and the benefits of being able to check for implementation before transferring are as useful for ERC20 tokens as they are for ERC721 and ERC1155.
```

Comment:
```suggestion
This EIP is meant to be minimal and straightforward. Adding EIP-165 to ERC20 is useful for a number of applications, and outside a minimal amount of code increasing contract size, carries no downside. The `safeTransfer` and `safeTransferFrom` functions are well recognized from ERC721 and ERC1155, and therefore keeping identical naming conventions is reasonable, and the benefits of being able to check for implementation before transferring are as useful for ERC20 tokens as they are for ERC721 and ERC1155.
```

Reply:
unless there's a strong need, I think this should be left as-is

Reply:
Just regular proofreading.
I will mostly skip it from next time, unless absolutely necessary!

Reply:
@wschwab You think it should be left as is because you believe that "...and outside a minimal amount..." is more correct, or for some other reason?

FWIW, I agree with @mryalamanchi that "...and outside of a minimal amount..." is grammatically correct.

Reply:
I;m of the opinion that my version was closer to common spoken parlance, though that's subjective, and once I've got multiple opinions asking me to change it, I'll change it in the next push.

Reply:
Perhaps I'm just getting old and not keeping up with language evolution.  😖  We could do a poll of teen aged girls to see which sounds more correct to them, since they are the dominant drivers of language evolution in human society.  😄

Reply:
I was coming at it from the POV of regular proofreading. Even a small preposition can sometimes make a huge diff (remember those tricky word problems to be written in code?).